packages/git/src/browser/git-contribution.ts

```diff
@@ -456,19 +467,26 @@ export class GitContribution implements CommandContribution, MenuContribution, T
             tooltip: GIT_COMMANDS.OPEN_CHANGES.label
         });
 
-        const registerItem = (item: Mutable<TabBarToolbarItem>) => {
+        const registerItem = (item: Mutable<TabBarToolbarItem>, whenRepositorySelected: boolean = true) => {
             const commandId = item.command;
             const id = '__git.tabbar.toolbar.' + commandId;
             const command = this.commands.getCommand(commandId);
             this.commands.registerCommand({ id, iconClass: command && command.iconClass }, {
                 execute: (widget, ...args) => widget instanceof ScmWidget && this.commands.executeCommand(commandId, ...args),
                 isEnabled: (widget, ...args) => widget instanceof ScmWidget && this.commands.isEnabled(commandId, ...args),
-                isVisible: (widget, ...args) => widget instanceof ScmWidget && this.commands.isVisible(commandId, ...args),
+                isVisible: (widget, ...args) =>
+                    widget instanceof ScmWidget &&
+                    this.commands.isVisible(commandId, ...args) &&
+                    !!this.repositoryProvider.selectedRepository === whenRepositorySelected,
```

Comment:
I wonder can we use `when` closure on a command instead? for example `!scmProvider` for missing scm provider at all or `scmProvider == git` for cases when scm provider is git

Reply:
BTW where do you pass it as `false`?

Reply:
Here https://github.com/theia-ide/theia/tree/8e4ceac6ff943c46586d62e25d7f847ff3f01c63#L485
When I look at it again, I think that it is total nonsense. I will change it, use the helper function as before just for the other repository related commands and register this Init Toolbar Item seperately.